coding-concepts.js

```diff
@@ -6,49 +6,63 @@
 // Then, uncomment the console.log() and verify the output. Briefly explain why your initial answer was correct or incorrect.
 
 // --------------------INSTRUCTOR EXAMPLE: What will this log?
-const colors = ["tangerine", "magenta", "lilac", "daffodil"]
+// const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
 // a) Your answer:
 // b) Verify and explain:
 
 // --------------------1) What will this log?
 
-const cohort = "LEARN 2022"
+// const cohort = "LEARN 2022"
 // console.log(cohort.length)
 
 // a) Your answer:
+    // Answer: This will return the number of elements in the array. 10 is the number of indexes it should return
 // b) Verify and explain:
+    // Answer: The .length property inspeted the indexes then returned the last index of the array plus one
 
 // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
 // a) Your answer:
+    // Answer: This should return the index of "o"
 // b) Verify and explain:
+    // Answer: The "o" is at the index of 4
 
 // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
 // a) Your answer:
+    // Answer: This should return "Ruby"
 // b) Verify and explain:
+    // "Ruby" is at the index of 1 
+
 
 // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+// const weekendDays = ["saturday", "sunday"]
+// // console.log(weekendDays.toUpperCase())
+
+// console.log("saturday, sunday".toUpperCase())
 
 // a) Your answer:
+    // This should return "SATURDAY, SUNDAY"
 // b) Verify and explain:
+    // Answer: Did not log what I thought it would because it did not have the correct function in the console log
```

Comment:
This is often a confusing error, but note what kind of error it says "Type error".  When we run into this error, it is actually telling us we are trying to use a built in method that does not work on this data type.  In this case, toUpperCase() is a string method, not an array method.